UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs

```diff
@@ -395,18 +463,105 @@ private void UpdateSeverity()
 		/// <summary>
 		/// Checks if the bodypart is damaged to a point where it can be gibbed from the body
 		/// </summary>
-		protected void CheckBodyPartIntigrity()
+		protected void CheckBodyPartIntigrity(float lastDamage)
 		{
-			if(Severity >= GibsOnSeverityLevel)
+			if(lastDamage >= cutDamgeThreshold)
+			{
+				ApplyCutSizeLogic();
+			}
+			if(currentCutSize >= BodyPartDisembowlLogicOnCutSize)
+			{
+				if(containBodyParts.Count != 0)
+				{
+					Disembowel();
+				}
+			}
+			if(Severity >= GibsOnSeverityLevel && lastDamage >= DamageThreshold)
 			{
 				DismemberBodyPartWithChance();
 			}
 		}
 
+
+		private void ApplyCutSizeLogic()
+		{
+			float chance = UnityEngine.Random.Range(0.0f, 1.0f);
+			float armorChanceModifer = cutChance + SelfArmor.DismembermentProtectionChance; //We use dismember protection chance because it's the most logical value.
+			if(chance > armorChanceModifer)
+			{
+				currentCutSize = currentCutSize.Next();
+			}
+		}
+
+		/// <summary>
+		/// Checks if the cut is big enough for the contained organs to escape.
+		/// If the cut isn't big enough or has failed a chance check, apply internal damage + bleeding.
+		/// </summary>
+		private void Disembowel()
+		{
+			BodyPart randomBodyPart = ContainBodyParts.GetRandom();
+			BodyPart randomCustomBodyPart = OptionalOrgans.GetRandom();
+			if(currentCutSize >= BodyPartStorageContentsSpillOutOnCutSize)
+			{
+				float chance = UnityEngine.Random.Range(0.0f, 1.0f);
+				if(chance >= spillChanceWhenCutPresent)
+				{
+					randomBodyPart.RemoveFromBodyThis();
+					if(randomCustomBodyPart != null)
+					{
+						randomCustomBodyPart.RemoveFromBodyThis();
+					}
+				}
+				else
+				{
+					randomBodyPart.ApplyInternalDamage();
+					if(randomCustomBodyPart != null)
+					{
+						randomCustomBodyPart.ApplyInternalDamage();
+					}
+				}
+			}
+			else
+			{
+				randomBodyPart.ApplyInternalDamage();
+				if(randomCustomBodyPart != null)
+				{
+					randomCustomBodyPart.ApplyInternalDamage();
+				}
+			}
+		}
+
+		/// <summary>
+		/// Enables internal damage logic.
+		/// </summary>
+		private void ApplyInternalDamage()
+		{
+			float damageToTake = UnityEngine.Random.Range(MinMaxInternalBleedingValues.x, MinMaxInternalBleedingValues.y);
+			TakeDamage(null, damageToTake / 2, AttackType.Internal, DamageType.Brute, true, true, 0);
+			if(CanBleedInternally)
+			{
+				isBleedingInternally = true;
+			}
+		}
+
+		public virtual void InternalBleedingLogic()
+		{
+			float damageToTake = UnityEngine.Random.Range(MinMaxInternalBleedingValues.x, MinMaxInternalBleedingValues.y);
+			currentInternalBleedingDamage += damageToTake;
+			if(currentInternalBleedingDamage > maximumInternalBleedDamage)
+			{
+				BodyPart currentParent = GetParent();
+				if(currentParent != null)
+				{
+					currentParent.TakeDamage(null, damageToTake, AttackType.Internal, DamageType.Brute, false, false, 0);
```

Comment:
hummm balance, 

I think for internal bleeding It should only happen if  where the organ roughly takes about 50%  to 75%  damage

as for the effects,

idk what effect bleeding should have
need to discuss on Discord